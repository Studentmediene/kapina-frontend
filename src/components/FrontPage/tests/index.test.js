import React from 'react';
import { shallow } from 'enzyme';

import { FrontPage } from '../';

describe('<FrontPage />', () => {
  it('renders correctly', () => {
    const props = {
      posts: [],
      loading: false,
      error: false,
      pageNumber: 1,
      loadPosts: jest.fn(),
      hasLoaded: false,
    };

    const tree = shallow(<FrontPage {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly while loading', () => {
    const props = {
      posts: false,
      loading: true,
      error: false,
      pageNumber: 1,
      loadPosts: jest.fn(),
      hasLoaded: false,
    };

    const tree = shallow(<FrontPage {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly on error', () => {
    const props = {
      posts: false,
      loading: false,
      error: true,
      pageNumber: 1,
      loadPosts: jest.fn(),
      hasLoaded: false,
    };

    const tree = shallow(<FrontPage {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
