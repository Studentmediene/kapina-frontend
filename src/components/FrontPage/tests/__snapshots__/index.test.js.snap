// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FrontPage /> renders correctly 1`] = `
<div>
  <PostPreviewList
    posts={Array []}
  />
  <div
    className="loaderWrapper"
  >
    <LoadPostsButton
      loadPosts={[Function]}
    />
  </div>
</div>
`;

exports[`<FrontPage /> renders correctly on error 1`] = `
<div>
  Kunne ikke laste inn forsiden.
</div>
`;

exports[`<FrontPage /> renders correctly while loading 1`] = `
<div>
  <div
    className="loaderWrapper"
  >
    <Loader />
  </div>
</div>
`;
